src/microsoft-extensions-ai/abstraction-implementations/AbstractionImplementationExamples/ToolCalling.cs

```diff
@@ -0,0 +1,32 @@
+using System.ComponentModel;
+using Microsoft.Extensions.AI;
+
+public partial class AbstractionSamples
+{
+    public static async Task ToolCalling() 
+    {
+        [Description("Gets the weather")]
+        string GetWeather()
+        {
+            var r = new Random();
+            return r.NextDouble() > 0.5 ? "It's sunny" : "It's raining";
+        }
+
+        var chatOptions = new ChatOptions
+        {
+            Tools = [AIFunctionFactory.Create(GetWeather) ]
+        };
+
+        IChatClient client =
+            new ChatClientBuilder()
+                .UseFunctionInvocation()
```

Comment:
Same concern here about function calling with SampleChatClient. I'll stop commenting on it, but this goes for anywhere the two are used together.

Reply:
Tracking here - https://github.com/dotnet/ai-samples/issues/80